pkg/types/v2/metadata.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 Google Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleCloudPlatform/container-structure-test/pkg/drivers"
+	types "github.com/GoogleCloudPlatform/container-structure-test/pkg/types/unversioned"
+	"github.com/GoogleCloudPlatform/container-structure-test/pkg/utils"
+)
+
+type MetadataTest struct {
+	Env          []types.EnvVar `yaml:"env"`
+	ExposedPorts []string       `yaml:"exposedPorts"`
+	Entrypoint   *[]string      `yaml:"entrypoint"`
+	Cmd          *[]string      `yaml:"cmd"`
+	Workdir      string         `yaml:"workdir"`
+	Volumes      []string       `yaml:"volumes"`
+}
+
+func (mt MetadataTest) LogName() string {
+	return "Metadata Test"
+}
+
+func (mt MetadataTest) Run(driver drivers.Driver) *types.TestResult {
+	result := &types.TestResult{
+		Name: mt.LogName(),
```

Comment:
why does this one not look like the others? pass: true

Reply:
good catch